stores/src/main/java/dataBase/DataBaseStorePopulator.java

```diff
@@ -0,0 +1,149 @@
+                    " productRate DOUBLE, " +
+                    " category_id LONG NOT NULL, " +
+                    " FOREIGN KEY (category_id) REFERENCES CATEGORY (id), " +
+                    " PRIMARY KEY ( id ))";
+            statement.executeUpdate(sql2);
+
+            String sql3 = "SELECT id FROM CATEGORY ";
+            ResultSet rs3 = statement.executeQuery(sql3);
+            int count = 0;
+            while (rs3.next()) {
+                count++;
+            }
+//            System.out.println(count);
+
+            if (count > 1) {
+                fillCategoryDB();
+//
+            } else {
+                List<Category> categoryTable = new RandomStorePopulator().createListCategories();
+                for (Category category : categoryTable) {
+                    String SQL = "INSERT INTO  CATEGORY ( name)  " + "VALUES( ?)";
+                    preparedStatement = dataBaseConnect.connection().prepareStatement(SQL, Statement.RETURN_GENERATED_KEYS);
+                    preparedStatement.setString(1, category.getName());
+                    preparedStatement.executeUpdate();
+                    ResultSet rs = preparedStatement.getGeneratedKeys();
+                    rs.next();
+                    List<Product> productTable = RandomStorePopulator.createListProduct(category.getName());
+                    for (Product product : productTable) {
+                        String SQL1 = "INSERT INTO PRODUCT (productName, productPrice, productRate, category_id)" + "VALUES( ?, ?, ?, ?)";
+                        preparedStatement = dataBaseConnect.connection().prepareStatement(SQL1);
+                        preparedStatement.setString(1, product.getName());
+                        preparedStatement.setDouble(2, product.getPrice());
+                        preparedStatement.setDouble(3, product.getRate());
+                        preparedStatement.setLong(4, rs.getLong(1));
+                        preparedStatement.executeUpdate();
+                    }
+                }
+                fillCategoryDB();
+            }
+
+            statement.close();
+            dataBaseConnect.connection().close();
+        } catch (SQLException se) {
+            throw new RuntimeException("Database access error", se);
+        } catch (Exception e) {
+            throw new RuntimeException("Error", e);
+        } finally {
+            try {
+                if (statement != null) statement.close();
+            } catch (SQLException se2) {
+                throw new RuntimeException("Database access error", se2);
+            }
+            try {
+                if (dataBaseConnect.connection() != null) dataBaseConnect.connection().close();
+            } catch (SQLException se) {
+                throw new RuntimeException("Database access error", se);
+            }
+        }
+//        System.out.println(categoryListDB);
+        return categoryListDB;
+    }
+
+    private void fillCategoryDB() throws SQLException {
```

Comment:
fill is a bad method name, fill - is inserting/populating, in your case it is getCategoriesFromDB or so, and please make function return proper values.